servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingStreamingHttpRequest.java

```diff
@@ -49,14 +50,38 @@ default InputStream payloadBodyInputStream() {
 
     /**
      * Gets and deserializes the payload body.
+     * @deprecated Use {@link #payloadBody(HttpStreamingDeserializer)}.
      * @param deserializer The function that deserializes the underlying {@link BlockingIterable}.
      * @param <T> The resulting type of the deserialization operation.
      * @return The results of the deserialization operation.
      */
+    @Deprecated
     default <T> BlockingIterable<T> payloadBody(HttpDeserializer<T> deserializer) {
         return deserializer.deserialize(headers(), payloadBody());
     }
 
+    /**
+     * Gets and deserializes the payload body.
+     * @param deserializer The function that deserializes the underlying {@link BlockingIterable}.
+     * @param <T> The resulting type of the deserialization operation.
+     * @return The results of the deserialization operation.
+     */
+    <T> BlockingIterable<T> payloadBody(HttpStreamingDeserializer<T> deserializer);
+
+    /**
+     * Get the {@link HttpMessageBodyIterable} for this request.
+     * @return the {@link HttpMessageBodyIterable} for this request.
+     */
+    HttpMessageBodyIterable<Buffer> messageBody();
```

Comment:
continuation of https://github.com/apple/servicetalk/pull/1673#discussion_r683957514

@idelpivnitskiy @chemicL - note that I removed `BlockingStreamingHttpMessageBody` originally introduced in this PR in favor of `HttpMessageBodyIterable`. The motivation is the functional APIs such as TrailersTransformer doesn't lend themselves to the sequential programming style of traditional blocking APIs. I meant to deprecate the transform methods on the blocking request/response APIs but forgot (will push this change). The `BlockingIterable` types are still cumbersome to create but I think that can be improved over time (more utilities) and also we have a task to investigate removal of `BlockingIterable` (in favor of `ClosableIterable`).